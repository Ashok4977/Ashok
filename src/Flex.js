import React from 'react'
import classnames from 'classnames'
import responsive, {any, only} from './utils/responsive'

const anyDirection = any('row', 'column', 'row-reverse')
const anyJustify = any('start', 'end', 'center', 'between', 'around')
const anyItems = any('start', 'end', 'center', 'baseline', 'stretch')
const anyContent = any('start', 'end', 'center', 'between', 'around', 'stretch')
const anySelf = any('auto', 'start', 'end', 'center', 'baseline', 'stretch')

const Flex = props => {
  const {
    direction,
    wrap,
    justify,
    items,
    content,
    auto,
    shrink,
    self,
    itemEqual,
    children
  } = props
  return (
    <div className={classnames(
      'd-flex',
      responsive('flex-direction-', direction, anyDirection),
      responsive('flex-', wrap, any(true, false), v => (v ? 'wrap' : 'nowrap')),
      responsive('flex-justify-', justify, anyJustify),
      responsive('flex-items-', items, anyItems),
      responsive('flex-content-', content, anyContent),
    )}>
      {children}
    </div>
  )
}

const FlexItem = props => {
  const {
    auto,
    shrink,
    self,
    itemEqual,
    children
  } = props
  return (
    <div className={classnames(
      responsive('flex-auto', auto, only(true), v => ''),
      responsive('flex-shrink-', shrink, only(0)),
      responsive('flex-self-', self, anySelf),
      responsive('flex-item-equal', itemEqual, only(true))
    ) || undefined}>
      {children}
    </div>
  )
}

Flex.Item = FlexItem

export default Flex
export {FlexItem}
