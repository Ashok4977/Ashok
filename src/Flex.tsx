import styled from 'styled-components'
import Box, {BoxProps} from './Box'
import {ForwardRefComponent, IntrinsicElement} from './utils/polymorphic'

export type FlexProps = BoxProps

type FlexComponent = ForwardRefComponent<IntrinsicElement<typeof Box>, FlexProps>

const Flex = styled(Box)`` as FlexComponent

Flex.defaultProps = {
  display: 'flex'
}

export default Flex
