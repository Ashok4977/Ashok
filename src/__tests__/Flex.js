import React from 'react'
import Flex, {FlexItem} from '../Flex'
import renderer from 'react-test-renderer'

const render = component => renderer.create(component).toJSON()

it('Flex obeys direction', () => {
  expect(render(<Flex direction='row' />))
    .toEqual(render(<div className='d-flex flex-direction-row' />))
  expect(render(<Flex direction='column' />))
    .toEqual(render(<div className='d-flex flex-direction-column' />))
  expect(render(<Flex direction='row-reverse' />))
    .toEqual(render(<div className='d-flex flex-direction-row-reverse' />))
  expect(render(<Flex direction='invalid' />))
    .toEqual(render(<div className='d-flex' />))
})

it('Flex obeys responsive direction', () => {
  expect(render(<Flex direction={['column', null, null, 'row']} />))
    .toEqual(render(<div className='d-flex flex-direction-column flex-lg-direction-row' />))
  expect(render(<Flex direction={['column', 'row', null, null, 'row-reverse']} />))
    .toEqual(render(<div className='d-flex flex-direction-column flex-sm-direction-row flex-xl-direction-row-reverse' />))
})

it('Flex obeys wrap', () => {
  expect(render(<Flex wrap={true} />))
    .toEqual(render(<div className='d-flex flex-wrap' />))
  expect(render(<Flex wrap={false} />))
    .toEqual(render(<div className='d-flex flex-nowrap' />))
})

it('Flex obeys justify', () => {
  expect(render(<Flex justify='start' />))
    .toEqual(render(<div className='d-flex flex-justify-start' />))
  expect(render(<Flex justify='end' />))
    .toEqual(render(<div className='d-flex flex-justify-end' />))
})

it('FlexItem supports auto', () => {
  expect(render(<FlexItem auto />))
    .toEqual(render(<div className='flex-auto' />))
  expect(render(<FlexItem auto={false} />))
    .toEqual(render(<div />))
})
