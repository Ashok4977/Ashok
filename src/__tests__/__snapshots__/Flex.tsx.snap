// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex renders consistently 1`] = `
<div
  className=""
  display="flex"
/>
`;

exports[`Flex respects alignContent 1`] = `
<div
  className=""
  display="flex"
/>
`;

exports[`Flex respects alignItems 1`] = `
<div
  className=""
  display="flex"
/>
`;

exports[`Flex respects flexDirection 1`] = `
<div
  className=""
  display="flex"
/>
`;

exports[`Flex respects flexWrap 1`] = `
<div
  className=""
  display="flex"
/>
`;

exports[`Flex respects justifyContent 1`] = `
<div
  className=""
  display="flex"
/>
`;

exports[`Flex respects responsive display 1`] = `
<div
  className=""
  display={
    Array [
      "flex",
      "inline-flex",
    ]
  }
/>
`;
